pkg/oam/util/template.go

```diff
@@ -24,23 +24,24 @@ type Template struct {
 }
 
 // GetScopeGVK Get ScopeDefinition
-func GetScopeGVK(cli client.Client, dm discoverymapper.DiscoveryMapper,
+func GetScopeGVK(ctx context.Context, cli client.Reader, dm discoverymapper.DiscoveryMapper,
 	name string) (schema.GroupVersionKind, error) {
 	var gvk schema.GroupVersionKind
 	sd := new(v1alpha2.ScopeDefinition)
-	if err := cli.Get(context.Background(), client.ObjectKey{
-		Name: name,
-	}, sd); err != nil {
+	err := GetDefinition(ctx, cli, sd, name)
+	if err != nil {
 		return gvk, err
```

Comment:
normally we don't return the value when there is an error

Reply:
The logic in here, when got a not nil error from GetDefinition, the func would return an empty struct GVK